django/template/defaultfilters.py

```diff
@@ -430,7 +430,10 @@ def rjust(value, arg):
 @stringfilter
 def center(value, arg):
     """Center the value in a field of a given width."""
-    return value.center(int(arg))
+    width = int(arg)
+    if width < 0:
```

Comment:
We can also include the `width == 0` case:

```suggestion
    if width <= 0:
```